### PR TITLE
Fix exception caused by wrong bound function

### LINKED ISSUE #1176: Uncaught TypeError: Cannot read property 'path' of undefined
[Enter steps to reproduce below:]

1. Running a flask app.
2. Hit the link shown in the console http://localhost:5000.

**Atom Version**: 1.12.6
**Electron Version**: 1.4.10
**System**: linux 4.8.12-2-ARCH
**Thrown From**: [script](https://github.com/rgbkrk/atom-script) package, v3.12.0

### Stack Trace

Uncaught TypeError: Cannot read property 'path' of undefined

```
At /home/gabriel/.atom/packages/script/lib/link-paths.js:18

TypeError: Cannot read property 'path' of undefined
    at HTMLAnchorElement.<anonymous> (/home/gabriel/.atom/packages/script/lib/link-paths.js:18:15)
    at HTMLDivElement.dispatch (/usr/lib/atom/node_modules/jquery/dist/jquery.js:4435:9)
    at HTMLDivElement.elemData.handle (/usr/lib/atom/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
  2x -1:06.2.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -1:05.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -1:05.2.0 application:add-project-folder (div.panels-item)
     -0:58.1.0 command-palette:toggle (div.panels-item)
  2x -0:52.5.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:51.5.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:51.5.0 pane:save-items (div.panels-item)
     -0:36.7.0 tree-view:show-in-file-manager (span.name.icon.icon-file-directory)
     -0:18.9.0 fuzzy-finder:toggle-file-finder (atom-text-editor.editor.is-focused)
 25x -0:17.1.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -0:08.8.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -0:08.8.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -0:07.6.0 script:run (atom-text-editor.editor.is-focused)
     -0:06.9.0 fuzzy-finder:toggle-buffer-finder (atom-text-editor.editor.is-focused)
     -0:01.7.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -0:01.7.0 core:cancel (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "telemetryConsent": "limited",
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v0.4.6 (active)
atom-material-ui, v1.3.8 (active)
autocomplete-python, v1.8.20 (active)
script, v3.12.0 (active)
stacktrace, v0.0.2 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.45.0 (inactive)
base16-tomorrow-dark-theme, v1.3.0 (inactive)
base16-tomorrow-light-theme, v1.3.0 (inactive)
one-dark-ui, v1.6.2 (inactive)
one-light-ui, v1.6.2 (inactive)
one-dark-syntax, v1.5.0 (inactive)
one-light-syntax, v1.5.0 (inactive)
solarized-dark-syntax, v1.0.5 (inactive)
solarized-light-syntax, v1.0.5 (inactive)
about, v1.7.0 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.13.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.4 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.2 (active)
command-palette, v0.39.0 (active)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.202.2 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.158.8 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.1 (inactive)
settings-view, v0.243.1 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.4 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.2 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.210.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.0 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

### PR DESCRIPTION
**Pull request by lgeiger**
Fixes #1176

### CODE REVIEW

File: lib/link-paths.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
 'use babel';
 
-/* eslint-disable no-multi-str, prefer-const*/
+/* eslint-disable no-multi-str, prefer-const, func-names */
 let linkPaths;
-const regex = new RegExp('\
```
Comment (nixel2007) [inline lib/link-paths.js]: You can leave the multiline regexp if add (?x) flag at the beginning

File: lib/link-paths.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+// (?::(\d+))?            # Column number prefixed with a colon (optional)
+
 const template = '<a class="-linked-path" data-path="$1" data-line="$2" data-column="$3">$&</a>';
 
 export default linkPaths = lines => lines.replace(regex, template);
 
 linkPaths.listen = parentView =>
-  parentView.on('click', '.-linked-path', () => {
+  parentView.on('click', '.-linked-path', function () {
     const el = this;
```
Comment (rgbkrk) [inline lib/link-paths.js]: What is `this` bound to?

**Comment by rgbkrk** [comment]
Restarted the failing job.

File: lib/link-paths.js (hunk 2 above)
Comment (lgeiger) [inline lib/link-paths.js]: Actually, I don't know but it should be bound inside the function.

**Comment by rgbkrk** [comment]
:shipit:

File: lib/link-paths.js (hunk 1 above)
Comment (lgeiger) [inline lib/link-paths.js]: > if add (?x) flag at the beginning

I don't understand what you mean.

But I can change it back to a multiline regexp if you like. This change  has only stylistic purpose.

**Pull request merged**